text/0004-avoidance_input_output.md

```diff
@@ -0,0 +1,47 @@
+  * Start date: 2018-03-26
+  * Contributors: Dennis Mannhart <dennis@px4.io>, Martina Rivizzigno <martina@px4.io>, Christoph Tobler <christoph@px.4io>
+  * Related PR: [trajectory-msg](https://github.com/mavlink/mavlink/pull/856)
+  * Related document: [avoidance-proposal](https://docs.google.com/document/d/1BQp1a6yszl9f6LDrxrkKUDDGyXxBs5C86BzvJwVbRrU/edit#heading=h.huol20joi641)
+  
+# Summary
+
+A general message that provides information about the desired kinematics over some time horizon. Any avoidance- or trajectory-module can exploit this message to pass trajectory
+information to the autopilot. 
+
+  
+# Motivation
+
+As obstacle-detection and avoidance receive more attention among UAVs, the current lack of a general avoidance interface/protocol becomes a bottleneck in supporting existing avoidance modules. An offboard
+avoidance module requires an interface over which it receives or sends trajectory- and obstacle-information. This rcf is about sharing trajectory information between the autopilot
+and the offboard module.
+
+# Detailed Design
+
+The trajectory message should comply with following requirements:
+* support a range of trajectory representations
+* a well-defined base message that can be extended
+* serves as input to and output from any offboard module
+
+The core of the message are five kinematic points in space, which we call `Anchor-Points`. Each `Anchor-Point` consists of an array of 11 elements. The `type`-field defines the interpretation of the 11 indices that 
+applies to all 5 `Anchor-Points`. 
+The first two suggested types are:
+* WAYPOINTS: Each `Anchor-Point` corresponds to a waypoint.
+  *	`[0-2]`: position
+  * `[3-5]`: velocity
+  * `[6-8]`: acceleration
+  * `[9]`  : yaw
+  * `[10]` : yaw-speed
+* BEZIER: `Anchor-Points` serve as spline parameters.
+  * `[0-2]`:  Bezier point
+  * `[3]`: Time horizon 
```

Comment:
https://github.com/mavlink/mavlink/pull/856/files has yaw and yawrate fields here which I think make sense

FYI: @Stifael @ChristophTobler